airflow/decorators/base.py

```diff
@@ -202,7 +202,16 @@ def task_decorator_factory(
     # try to infer from  type annotation
     if python_callable and multiple_outputs is None:
         sig = signature(python_callable).return_annotation
-        ttype = getattr(sig, "__origin__", None)
+
+        # If the return type annotation is already the builtins ``dict`` type, use it for the inference.
+        if sig == dict:
+            ttype = sig
```

Comment:
Does this cover `typing.Dict` and e.g. `typing.Dict[str, typing.Any]`?

Reply:
That particular check handles the simple `def func(...) -> dict:` case.  The remaining checks handle the `typing.Dict` and `typing.Dict[str, typing.Any]` cases.

Reply:
I updated the unit tests for some different flavors of typing too if it helps.